src/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -837,10 +837,14 @@ void activity_classifier::init_train(
     for (size_t i = 1; i < gpu_names.size(); ++i) {
       gpu_names_string += ", " + gpu_names[i];
     }
-    logprogress_stream << "Using "
-                       << (gpu_names.size() > 1 ? "GPUs" : "GPU")
-                       << " to create model ("
-                       << gpu_names_string << ")";
+    if (gpu_names_string.find("AMD") != std::string::npos) {
+      logprogress_stream << "Using " << (gpu_names.size() > 1 ? "GPUs" : "GPU")
+                         << " to create model (" << gpu_names_string << ")";
```

Comment:
For this message and the next message (line 844), would it make more sense to print the `gpu_names_string` between the number of GPUs and the `GPUs`/`GPU` string, rather than in parentheses at the end?

Reply:
Tensorflow returns GPU names like device:GPU:0, device:GPU:1, etc. Do we want these names? I feel like these names aren't good enough to be a part of the verbose.

And the lines 841 is what was being printed for mps before as well (no changes made by me) and the lines from 844 match what we had as output in MXNet (Python).

Reply:
Could you list what the Python toolkit currently prints (for Linux GPU)? And what the C++ version prints for Mac GPU and Linux GPU with your changes?

We should do what's sensible first and foremost, not necessarily match precisely what the Python version did. It is after all a new major version number :smile:

Reply:
When MXNet has GPUs to train it gives the following statements - 
`Using 1 GPU to create model (CUDA)` or `Using 4 GPUs to create model (CUDA)`
MPS (which uses C++ implementation) prints 
`Using GPU to create model (AMD Radeon Pro 560X)`
My change will make tensorflow print exactly what MXNet printed. Let me know if I should change it. 
But the device names in TF are given like `[PhysicalDevice(name=u'/physical_device:CPU:0', device_type=u'CPU')]`. Let me know if anyone has better ideas for verbose.